From ed04990da6e033c4429dc8710e300e9d7532d0ec Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Fri, 18 Sep 2020 17:50:10 +0300
Subject: [PATCH] [SW-0] goya2: uart: add uart padsel conf

UART pad select configuration added for preboot
and ppboot goya2 asic platform.

Following are the pad/pinmux settings for uart,

GPIO_08 (UART 0 RXD, ALT FUNC1)
GPIO_09 (UART 0 TXD, ALT FUNC2)

These settings are already done in BTL code. So
these are not required in flow where BTL is run.

Only in non-BTL flows(e.g. JTAG, secure boot) these
settings are required.

Change-Id: I86d1feaea3ca15c346a5700371c92092720b2f23
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/goya2_arc/asic.c b/goya2_arc/asic.c
index 379478c..00fb888 100644
--- a/goya2_arc/asic.c
+++ b/goya2_arc/asic.c
@@ -28,6 +28,8 @@
 	u32 uart_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_CTRL_BASE,
 						HL_PLL_DIV3);
 
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
 	preboot_uart_init(uart_clk_freq / HL_UART_BAUDRATE);
 }
 
diff --git a/goya2_arc_gnu/asic.c b/goya2_arc_gnu/asic.c
index 6d3fa5e..2164113 100644
--- a/goya2_arc_gnu/asic.c
+++ b/goya2_arc_gnu/asic.c
@@ -30,6 +30,8 @@
 	u32 uart_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_CTRL_BASE,
 						HL_PLL_DIV3);
 
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
 	preboot_uart_init(uart_clk_freq / HL_UART_BAUDRATE);
 }
 
